### ISSUE TITLE
Minimal resolution of overlapping cues is difficult

### ISSUE THREAD

**Issue by rgov** [w3c/webvtt#284]
> If there is a position to which the boxes in boxes can be moved while maintaining the relative positions of the boxes in boxes to each other such that none of the boxes in boxes would overlap any of the boxes in output, and all the boxes in boxes would be within the video’s rendering area, then move the boxes in boxes to the closest such position to their current position,

The spec does not propose an algorithm for doing this, and I suspect that it is equivalent to implementing a constraint solver for systems of linear inequalities. (I'm not against this; if a constraint solver is the way create the best layout, so it is.)

It's not difficult to display a new cue and find a non-overlapping place for it (or, whether there is no such place). But figuring out how to minimally move around existing cues to make room for a new one is nontrivial.

**Comment by fsoder** [comment]
Note that the above quote does not call for any "boxes in output" to be moved - just "boxes in boxes" (i.e the "new" cue being processed). So your second (non-quote) paragraph is what applies.

**Comment by zcorpan** [comment]
Yes, what @fsoder said. Is that still too difficult, or is it fine? I'm happy to change the algorithm to be dumber, e.g. just moving upwards, then downwards, then giving up (and analogously for vertical cues), or something.

**Comment by rgov** [comment]
Is moving up first correct? This is Safari's current behavior and there are instances where it has made captions hard to understand (making viewer read left to right, bottom to top).

**Comment by zcorpan** [comment]
No the current spec says to put it in the closest position and has some rule when there are multiple such positions. My point is that we can change it to match what you want to implement, or what people think is best for the user, or good enough, etc.

**Comment by foolip** [comment]
I think the overlap avoidance logic ought to be the same for line-positioned and percentage-positioned queues, which should simplify this.
